jc/parsers/srt.py

```diff
@@ -0,0 +1,204 @@
+
+
+class info():
+    """Provides parser metadata (version, author, etc.)"""
+    version = '1.0'
+    description = 'SRT file parser'
+    author = 'Mark Rotner'
+    author_email = 'rotner.mr@gmail.com'
+    compatible = ['linux', 'darwin', 'cygwin', 'win32', 'aix', 'freebsd']
+    tags = ['standard', 'file', 'string']
+
+
+__version__ = info.version
+
+
+def _process(proc_data):
+    """
+    Final processing to conform to the schema.
+
+    Parameters:
+
+        proc_data:  (Dictionary) raw structured data to process
+
+    Returns:
+
+        List of Dictionaries representing an SRT document.
+    """
+
+    int_list = {'index'}
+    timestamp_list = {"start", "end"}
+    timestamp_int_list = {"hours", "minutes", "seconds", "milliseconds"}
+
+    for entry in proc_data:
+        # Converting {"index"} to int.
+        for key in entry:
+            if key in int_list:
+                entry[key] = jc.utils.convert_to_int(entry[key])
+
+            # Converting {"hours", "minutes", "seconds", "milliseconds"} to int.
+            if key in timestamp_list:
+                timestamp = entry[key]
+                for timestamp_key in timestamp:
+                    if timestamp_key in timestamp_int_list:
+                        timestamp[timestamp_key] = jc.utils.convert_to_int(timestamp[timestamp_key])
+
+    return proc_data
+
+
+def parse_timestamp(timestamp: str):
+    """
+    timestamp: "hours:minutes:seconds,milliseconds" --->
+    {
+        "hours": "hours",
+        "minutes": "minutes",
+        "seconds": "seconds",
+        "milliseconds": "milliseconds",
+        "timestamp": "hours:minutes:seconds,milliseconds"
+    }
+    """
+
+    hours, minutes, seconds_milliseconds = timestamp.split(':')
+    seconds, milliseconds = seconds_milliseconds.split(',')
+    return {
+        "hours": hours,
+        "minutes": minutes,
+        "seconds": seconds,
+        "milliseconds": milliseconds,
+        "timestamp": timestamp
+    }
+
+
+def parse(data: str, raw: bool = False, quiet: bool = False):
+    """
+    Main text parsing function
+
+    Parameters:
+
+        data:        (string)  text data to parse
+        raw:         (boolean) unprocessed output if True
+        quiet:       (boolean) suppress warning messages if True
+
+    Returns:
+
+        Dictionary. Raw or processed structured data.
+    """
+    jc.utils.compatibility(__name__, info.compatible, quiet)
+    jc.utils.input_type_check(data)
+
+    raw_output = []
+    if not jc.utils.has_data(data):
+        return raw_output
+
+    subtitle = {}
+    lines = list(filter(None, data.splitlines()))
+    for line in lines:
+        line = line.strip()
+        if line.isdigit():  # Reached a new subtitle.
+            if subtitle:
+                raw_output.append(subtitle)
+                subtitle = {}
+            subtitle['index'] = line
+            continue
+
+        if '-->' in line:  # Start and end time of the subtitle separated by –> characters.

```

Comment:
I wonder if we should tighten this up a bit? I really can't think of why subtitles would have `-->` in them, so this is probably not a real issue, but I was thinking maybe we should use a regex that matches the `00:02:16,612 --> 00:02:19,376` pattern instead. Something like:

```
^\d\d:\d\d:\d\d,\d+ --> \d\d:\d\d:\d\d,\d+$
```